tesseract_task_composer/include/tesseract_task_composer/nodes/raster_motion_task.hpp

```diff
@@ -0,0 +1,336 @@
+/**
+ * @file raster_motion_task.hpp
+ * @brief Raster motion task with transitions
+ *
+ * @author Levi Armstrong
+ * @date July 29. 2022
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2022, Levi Armstrong
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TESSERACT_TASK_COMPOSER_RASTER_MOTION_TASK_HPP
+#define TESSERACT_TASK_COMPOSER_RASTER_MOTION_TASK_HPP
+
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <console_bridge/console.h>
+#include <boost/serialization/string.hpp>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+#include <tesseract_common/timer.h>
+
+#include <tesseract_task_composer/task_composer_task.h>
+#include <tesseract_common/any_poly.h>
+
+#include <tesseract_task_composer/task_composer_future.h>
+#include <tesseract_task_composer/task_composer_executor.h>
+#include <tesseract_task_composer/nodes/start_task.h>
+#include <tesseract_task_composer/nodes/update_start_and_end_state_task.h>
+#include <tesseract_task_composer/nodes/update_end_state_task.h>
+#include <tesseract_task_composer/nodes/update_start_state_task.h>
+#include <tesseract_command_language/composite_instruction.h>
+
+namespace tesseract_planning
+{
+/**
+ * @brief The RasterCtMotionTask class
+ * @details The required format is below.
+ *
+ * Composite
+ * {
+ *   Composite - from start
+ *   Composite - Raster segment
+ *   Composite - Transitions
+ *   Composite - Raster segment
+ *   Composite - Transitions
+ *   Composite - Raster segment
+ *   Composite - to end
+ * }
+ */
+template <typename FreespaceTaskType, typename RasterTaskType, typename TransitionTaskType>
+class RasterMotionTask : public TaskComposerTask
+{
+public:
+  using class_type = RasterMotionTask<FreespaceTaskType, RasterTaskType, TransitionTaskType>;
+
+  RasterMotionTask() = default;  // Required for serialization
+  // NOLINTNEXTLINE(performance-unnecessary-value-param)
+  RasterMotionTask(std::string input_key, std::string output_key, bool is_conditional, std::string name)
```

Comment:
You removed the defaults for is_conditional and name. Later when calling it you only set the is_conditional parameter. You need to fix this at one place and/or the other. See: https://github.com/tesseract-robotics/tesseract_planning/blob/c32e377106247d278f15a37a543c1361a52a32f6/tesseract_task_composer/include/tesseract_task_composer/nodes/raster_global_pipeline_task.hpp#L56

Reply:
It should just be using the default name. Did you run into an issue?

Reply:
Sorry, just now seeing this after I got the merge notification. The default setting of the names in the base class were removed, so it fails when you try and make a custom task node implementation and don't define the name. Also the implementation of the global planner explicitly does not include a setting of the name to the templated RasterTaskType in the above segment of code I referenced, so this will always fail.